util/reggen/gen_dv.py

```diff
@@ -72,7 +98,39 @@ def gen_core_file(outdir: str,
         yaml.dump(core_data, core_file, encoding='utf-8')
 
 
-def gen_dv(block: IpBlock, dv_base_prefix: str, outdir: str) -> int:
+def get_dv_base_names_objects(dv_base_names: List[str]) -> Dict[str, DvBaseNames]:
+    ''' Convert the dv_base_names from string to a list of DvBaseNames objects.'''
+    dv_base_names_dict = {}  # type: Dict[str, DvBaseNames]
+
+    if dv_base_names == ["dv_base"]:
+        return dv_base_names_dict
+
+    # Customerized dv_base_names comes with the format of a list:
+    # --dv-base-names block-1:type:entity-name block-2:type:entity-name
+    for item in dv_base_names:
+        try:
+            block, base_type, entity = item.split(":")
+        except ValueError:
+            log.error(f"Bad input arg: {item}")
+            sys.exit(1)
+
+        if block not in dv_base_names_dict:
+            dv_base_names_dict[block] = DvBaseNames()
+        block_dv_base_names = dv_base_names_dict[block]
+        block_dv_base_names.set_entity(base_type, entity)
```

Comment:
With defaultdict, these can be reduced to:
```python
        dv_base_names_dict[block].set_entity(base_type, entity)
```

Reply:
oh nice, this is fancy!